examples/plot_glm_retinal_ganglia.py

```diff
@@ -0,0 +1,270 @@
+#
+# We can also assume that their is a non-linear function governing
+# the underlying the firing patterns. Concreately, we can write down as
+# :math:`y = q(\beta_0 + X \beta) + \epsilon`, and
+# :math:`y \sim \text{Poiss}(\beta_0 + X \beta)`.
+# Here we use :math:`exp()` as the inverse link function :math:`q`.
+# We can use the :class:`GLM` class to predict the parameters.
+
+# create possion GLM instance
+glm_poisson = GLM(distr='poisson',
+                  alpha=0.05,
+                  learning_rate=1.0,
+                  score_metric='pseudo_R2',
+                  reg_lambda=1e-7)
+glm_poisson.fit(Xdsgn, spikes_binned)
+spikes_pred_poissonGLM = glm_poisson.predict(Xdsgn)
+
+#############################################################################
+# **Adding spikes history for predicting spike counts**
+#
+# We can even further predict the spikes by concatenating the spikes history.
+#
+# .. note::
+#    The spike-history portion of the design
+#    matrix had better be shifted so that we aren't allowed to
+#    use the spike count on this time bin to predict itself!
+
+n_t_filt = 25  # same as before, stimulation history
+n_t_hist = 20  # spikes history
+
+# using both stimulation history and spikes history
+spikes_padded = np.pad(spikes_binned, (n_t_hist, 0))
+
+Xstim = hankel(stim_padded[:-n_t_filt + 1], stim[-n_t_filt:])
+Xspikes = hankel(spikes_padded[:-n_t_hist], stim[-n_t_hist:])
+Xdsgn_hist = np.hstack((Xstim, Xspikes))  # design matrix with spikes history
+
+# Now, we are ready to fit Poisson GLM with spikes history.
+# create possion GLM instance
+glm_poisson_hist = GLM(distr='poisson', alpha=0.05,
+                       learning_rate=1.0,
+                       score_metric='pseudo_R2',
+                       reg_lambda=1e-7)
+
+# fitting to a design matrix with spikes history
+glm_poisson_hist.fit(Xdsgn_hist, spikes_binned)
+
+# predict spike counts
+spikes_pred_poissonGLM_hist = glm_poisson_hist.predict(Xdsgn_hist)
+
+#############################################################################
+# **Putting all together**
+#
+# We are plotting the prediction of spike counts using
+# linear Gaussian GLM with offset, Poisson GLM, and
+# Poisson GLM with spikes history for one second.
+
+plt.figure()
+markerline, _, _ = plt.stem(t_sample, spikes_binned[sample_idx])
+markerline.set_markerfacecolor('none')
+plt.plot(t_sample, spikes_pred_lgGLM_offset[sample_idx],
+         color='gold', linewidth=2, label='lgGLM with offset')
+plt.plot(t_sample, spikes_pred_poissonGLM[sample_idx],
+         color='green', linewidth=2, label='poissonGLM')
+plt.plot(t_sample, spikes_pred_poissonGLM_hist[sample_idx],
+         color='red', linewidth=2, label='poissonGLM_hist')
+plt.xlim([0., tmax])
+plt.title('Spike count prediction')
+plt.xlabel('Time (sec)')
+plt.ylabel('Binned Spike Counts')
+plt.legend()
+plt.show()
+
+# performance of the fitted models
+mse_lgGLM_offset = np.mean((
+    spikes_binned - spikes_pred_lgGLM_offset) ** 2)
```

Comment:
can we use `glm.score` here?

Reply:
Yes, it just gave different scores. I will do further check on this.